data/base/endpoint.php

```diff
@@ -71,20 +72,45 @@ public function __construct( $controller ) {
 	/**
 	 * Get base route.
 	 *
-	 * Removing 'index' from endpoint.
-	 *
 	 * @return string
 	 */
 	public function get_base_route() {
-		$endpoint_name = $this->get_name();
+		$endpoint_public_name = $this->get_name_public();
+
+		return '/' . $this->controller->get_rest_base() . rtrim( $endpoint_public_name, '/' );
+	}
 
-		// TODO: Allow this only for internal routes.
-		// TODO: Make difference between internal and external endpoints.
-		if ( 'index' === $endpoint_name ) {
-			$endpoint_name = '';
+	/**
+	 * Get command public.
+	 *
+	 * Convert endpoint to command name ( without index ).
+	 *
+	 * Returns '/{name}/' or '/' ( for index ).
+	 *
+	 * @return string
+	 */
+	public function get_name_public() {
+		$endpoint_public_name = '';
+
+		if ( ! $this instanceof Index ) {
```

Comment:
The parent should know about his children, please find another solution.

Reply:
Handled: https://github.com/elementor/elementor/pull/12820/commits/b4cdcd462691d96b8f0b3424c8e636315098bab2